pkg/resourceinterpreter/default/native/retain.go

```diff
@@ -160,3 +162,28 @@ func retainWorkloadReplicas(desired, observed *unstructured.Unstructured) (*unst
 
 	return desired, nil
 }
+
+func retainSecretServiceAccountToken(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, err error) {
+	if secretType, exists, _ := unstructured.NestedString(desired.Object, "type"); exists && secretType == string(corev1.SecretTypeServiceAccountToken) {
+		// retain service-account.uid which is a unique per cluster
+		serviceAccountUIDPath := []string{"metadata", "annotations", corev1.ServiceAccountUIDKey}
+		uid, _, err := unstructured.NestedString(observed.Object, serviceAccountUIDPath...)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get %s from desired.Object: %+v", corev1.ServiceAccountUIDKey, err)
```

Comment:
Given this error, it could lead to sync failure.
Can we assume the `.metadata.annotations.kubernetes.io/service-account.uid` must exist?

I guess the logic should be:
- raise the error in case there is a concrete exception, like trying to find the value with a wrong path
- skip the retention if the uuid not found(not populate yet).

Reply:
I make a test. If the annotation `kubernetes.io/service-account.uid` of the secret in the member cluster is deleted, an empty annotation `kubernetes.io/service-account.uid` is added during retain. Like this:

![image](https://github.com/karmada-io/karmada/assets/78244870/f06829f7-5641-4d43-bf0b-5b8387aa23dc)

No error will occur. I still recommend to process the returned bool value. If the bool refers to false, there is no need to SetNestedField.